.github/workflows/deploy-staging.yml

```diff
@@ -0,0 +1,122 @@
+name: Deploy to Staging
+
+on:
+  push:
+    branches:
+      - main
+      - athai/github-deploy
+  workflow_dispatch: # Allow manual deployment
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    env:
+      NODE_VERSION: "20.6.0"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: "npm"
+          cache-dependency-path: |
+            server/package-lock.json
+            client/package-lock.json
+
+      - name: Install server dependencies
+        run: |
+          cd server
+          npm ci
+
+      - name: Install client dependencies
+        run: |
+          cd client
+          npm ci
+
+      - name: Validate build (optional)
+        run: |
+          cd client
+          npm run build:staging
+          echo "✅ Client build validation completed"
+
+      - name: Deploy to Lightsail instance
+        run: |
+          # Install SSH key
+          mkdir -p ~/.ssh
+          echo "${{ secrets.LIGHTSAIL_SSH_KEY }}" > ~/.ssh/lightsail_key
+          chmod 600 ~/.ssh/lightsail_key
+
+          # Deploy using git pull and rebuild
+          ssh -i ~/.ssh/lightsail_key -o StrictHostKeyChecking=no \
+            bitnami@${{ vars.LIGHTSAIL_HOST }} << 'ENDSSH'
+            
+            set -e  # Exit on any error
+            
+            echo "🚀 Starting deployment..."
+            
+            # Navigate to the repository
+            cd ~/bolt-gameserver-sample
+            
+            # Fetch latest changes
+            echo "🔄 Fetching latest code..."
+            git fetch origin
+            git reset --hard origin/${{ github.ref_name }}
+            
+            # Install server dependencies
+            echo "📦 Installing server dependencies..."
+            cd server
+            npm ci
+            
+            # Build client
+            echo "🏗️  Building client..."
+            cd ../client
+            npm ci
+            npm run build:staging
```

Comment:
There should be a step after this to restart nginx per the documented manual deploy guide